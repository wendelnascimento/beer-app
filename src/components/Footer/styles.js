import styled from 'styled-components';

export const Wrapper = styled.footer`
  padding: 16px;
  color: #cccccc;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  border-top: 1px solid #fafafa;
`;
