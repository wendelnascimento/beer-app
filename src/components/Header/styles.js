import styled from 'styled-components';

export const Wrapper = styled.header`
  max-width: 100vw;
  background: #fff;
  padding: 16px;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  border-bottom: 1px solid #fafafa;
`;

export const Title = styled.h1`
  font-weight: 300;
  margin: 0;
`;
