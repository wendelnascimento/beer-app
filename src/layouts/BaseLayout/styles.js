import styled from 'styled-components';

export const Wrapper = styled.div`
  max-width: 100vw;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
`;
