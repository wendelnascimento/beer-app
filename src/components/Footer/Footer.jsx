import React from 'react';

import { Wrapper } from './styles';

const Footer = () => (
  <Wrapper>
    <span>
      Beer App
      {' '}
      <span role="img" aria-label="Beer">
        🍺
      </span>
      {' '}
      - ©
      {' '}
      {new Date().getFullYear()}
    </span>
  </Wrapper>
);

export default Footer;
