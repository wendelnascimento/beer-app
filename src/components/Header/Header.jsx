import React from 'react';

import { Wrapper, Title } from './styles';

const Header = () => (
  <Wrapper>
    <Title>
      Beer App
      <span role="img" aria-label="Beer">
        🍺
      </span>
    </Title>
  </Wrapper>
);

export default Header;
