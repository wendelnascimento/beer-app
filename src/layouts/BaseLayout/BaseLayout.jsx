import React from 'react';
import PropTypes from 'prop-types';

import Header from '../../components/Header';
import Footer from '../../components/Footer';

import { Wrapper } from './styles';

const BaseLayout = ({ children }) => (
  <Wrapper>
    <Header />
    {children}
    <Footer />
  </Wrapper>
);

BaseLayout.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
};

export default BaseLayout;
