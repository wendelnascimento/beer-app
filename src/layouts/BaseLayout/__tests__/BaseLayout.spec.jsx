import React from 'react';
import { shallow, mount } from 'enzyme';

import BaseLayout from '../BaseLayout';
import Header from '../../../components/Header';
import Footer from '../../../components/Footer';

describe('<BaseLayout />', () => {
  it('does render properly', () => {
    shallow(
      <BaseLayout>
        <h1>Test</h1>
      </BaseLayout>,
    );
  });

  it('matches stored snapshot', () => {
    const wrapper = mount(
      <BaseLayout>
        <h1>Test</h1>
      </BaseLayout>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('does have <Header /> component inside of it', () => {
    const wrapper = mount(
      <BaseLayout>
        <h1>Test</h1>
      </BaseLayout>,
    );

    expect(wrapper.contains(Header)).toEqual(true);
  });

  it('does have <Footer /> component inside of it', () => {
    const wrapper = mount(
      <BaseLayout>
        <h1>Test</h1>
      </BaseLayout>,
    );

    expect(wrapper.contains(Footer)).toEqual(true);
  });

  it('does render children between <Header /> and <Footer />', () => {
    const wrapper = mount(
      <BaseLayout>
        <h1>Test</h1>
      </BaseLayout>,
    );

    expect(
      wrapper
        .find('h1')
        .at(1)
        .text(),
    ).toBe('Test');
    expect(wrapper.find('Header + h1')).toHaveLength(1);
    expect(wrapper.find('h1 + Footer')).toHaveLength(1);
  });
});
