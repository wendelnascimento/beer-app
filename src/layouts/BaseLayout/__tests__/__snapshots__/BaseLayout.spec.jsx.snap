// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseLayout /> matches stored snapshot 1`] = `
.c1 {
  max-width: 100vw;
  background: #fff;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  border-bottom: 1px solid #fafafa;
}

.c2 {
  font-weight: 300;
  margin: 0;
}

.c3 {
  padding: 16px;
  color: #cccccc;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  border-top: 1px solid #fafafa;
}

.c0 {
  max-width: 100vw;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<BaseLayout>
  <styled.div>
    <div
      className="c0"
    >
      <Header>
        <styled.header>
          <header
            className="c1"
          >
            <styled.h1>
              <h1
                className="c2"
              >
                Beer App
                <span
                  aria-label="Beer"
                  role="img"
                >
                  üç∫
                </span>
              </h1>
            </styled.h1>
          </header>
        </styled.header>
      </Header>
      <h1>
        Test
      </h1>
      <Footer>
        <styled.footer>
          <footer
            className="c3"
          >
            <span>
              Beer App
               
              <span
                aria-label="Beer"
                role="img"
              >
                üç∫
              </span>
               
              - ¬©
               
              2020
            </span>
          </footer>
        </styled.footer>
      </Footer>
    </div>
  </styled.div>
</BaseLayout>
`;
